src/ElfUtils/ElfFile.cpp

```diff
@@ -175,40 +175,34 @@ ErrorMessageOr<ModuleSymbols> ElfFileImpl<ElfT>::LoadSymbols() {
 template <typename ElfT>
 ErrorMessageOr<uint64_t> ElfFileImpl<ElfT>::GetLoadBias() const {
   const llvm::object::ELFFile<ElfT>* elf_file = object_file_->getELFFile();
-
-  uint64_t min_vaddr = UINT64_MAX;
-  bool pt_load_found = false;
   llvm::Expected<typename ElfT::PhdrRange> range = elf_file->program_headers();
 
   if (!range) {
-    std::string error = absl::StrFormat(
-        "Unable to get load bias of ELF file: \"%s\". No program headers "
-        "found.",
-        file_path_.string());
+    std::string error =
+        absl::StrFormat("Unable to get load bias of ELF file: \"%s\". No program headers found.",
+                        file_path_.string());
     ERROR("%s", error.c_str());
     return ErrorMessage(std::move(error));
   }
 
+  // Find the executable segment and calculate the load bias based on that segment.
```

Comment:
I think we briefly mentioned it yesterday: We assume a single executable segment? Could you clarify?

Reply:
In practice we have one, but there is nothing wrong from the elf-file format point of view to have more then one, ideally we want the one with .text segment, but there is a use case where this is not possible, debug symbols file does not have any executable sections. It is however keeps the 0 sized executable segment, I am assuming specifically so that debug tools could calculate load_bias for symbol adresses.